File: Data Structures & Algorithms/sqrtx/submission-2.java
class Solution {
    public int mySqrt(int x) {
        long l = 2, r = x;
        while(l <= r){
            long m = l + (r - l) / 2;
            if(m * m == x){
                return (int) m;
            } 
            else if(m * m > x){
                r = m - 1;
            }
            else{
                l = m + 1;
            }
        }
        return (int)r;
    }
}

File: Data Structures & Algorithms/valid-palindrome-ii/submission-2.cpp
class Solution {
public:
    bool validPalindrome(string s) {
        string temp;
        for(int i = 0; i < s.length() ; i++){
            if(isalnum(s[i])){
                temp += tolower(s[i]);
            }
        }
        int l = 0, r = temp.length() - 1;
        while(l < r){
            if(temp[l] != temp[r]){
                string s1 = temp.substr(l, r - l);
                string s2 = temp.substr(l + 1, r - l);
                bool res = helper(s1);
                res == false ? res = helper(s2) : res;
                return res;
            }
            l++; r--;
        }
        return true;
}
    bool helper(string s){
        int l = 0, r = s.length() - 1;
        while(l< r){
            if(s[l] != s[r]){
                return false;
            }
            l++; r--;
        }
        return true;
    }
};